### PR TITLE
Hookup edit profile mutation on identity edit view and add profile discoverability setting.

### PR DESCRIPTION
**Pull request by avclarke**

### CODE REVIEW

File: packages/daemon/src/context/ContextMutations.js [hunk 1]
```diff
@@ -356,25 +356,54 @@ export default class ContextMutations {
...
+  async setUserProfileVisibility(
+    userID: string,
+    visibile: boolean,
+  ): Promise<void> {
+    const user = this._context.openVault.identities.getOwnUser(userID)
+    if (!user) throw new Error('User not found')
+    if (user.privateProfile !== visibile) {
+      user.privateProfile = visibile
+      await this._context.openVault.save()
+      await user.publicFeed.publishJSON(
```
Comment (howleysv) [inline packages/daemon/src/context/ContextMutations.js]: With the new event system, you can fire an event with `{type: 'user_changed', change: 'profile', ...}` and the feed update & vault save will be handled automatically

File: packages/launcher/src/renderer/launcher/identities/IdentityEditModal.js [hunk 2]
```diff
@@ -4,15 +4,23 @@ import React, { Component } from 'react'
 import { Text, TextField, Switch } from '@morpheus-ui/core'
 import styled from 'styled-components/native'
 import { type FormSubmitPayload } from '@morpheus-ui/forms'
+import { createFragmentContainer, graphql, commitMutation } from 'react-relay'
 
+import { EnvironmentContext } from '../RelayEnvironment'
 import FormModalView from '../../UIComponents/FormModalView'
-import { type User } from './IdentitiesView'
+import OwnUserIdentity from './__generated__/IdentityEditModal_ownUserIdentity.graphql.js'
```
Comment (PaulLeCam) [inline packages/launcher/src/renderer/launcher/identities/IdentityEditModal.js]: The type is exported as a named export so the import should be something like: `import type { IdentityEditModal_ownUserIdentity as OwnUserIdentity } from './__generated__/IdentityEditModal_ownUserIdentity.graphql'`

**Comment by howleysv** [comment]
I'm not sure if having "discoverability" be a toggle rather than a fuse sets the right expectation. Once you write it once to swarm, it's basically there forever.

File: packages/daemon/src/context/ContextMutations.js (hunk 1 above)
Comment (avclarke) [reply packages/daemon/src/context/ContextMutations.js]: nice, will update

File: packages/launcher/src/renderer/launcher/onboarding/OnboardIdentityView.js [hunk 3]
```diff
@@ -134,7 +143,12 @@ export default class OnboardIdentityView extends Component<Props, State> {
                 />
               </Column>
               <Column>
-                <Switch label="Make my name discoverable" name="discoverable" />
+                <Switch
+                  defaultValue={true}
```
Comment (PaulLeCam) [inline packages/launcher/src/renderer/launcher/onboarding/OnboardIdentityView.js]: I don't think this switch should be on by default

Comment (avclarke) [reply packages/launcher/src/renderer/launcher/onboarding/OnboardIdentityView.js]: I feel like most people won't mind having a public profile and a lot will overlook this setting during onboarding. If we have it false by default I think majority of users will end up having no name and will be a bit of a poor experience?

Comment (howleysv) [reply packages/launcher/src/renderer/launcher/onboarding/OnboardIdentityView.js]: "discoverable" is a bit of a misnomer in any case. To me it implies being part of a searchable index, whereas in reality you would have to work pretty hard to "discover" anyone, let alone the person you're actually looking for

Comment (avclarke) [reply packages/launcher/src/renderer/launcher/onboarding/OnboardIdentityView.js]: Yea good point should probably use some better messaging there

Comment (PaulLeCam) [reply packages/launcher/src/renderer/launcher/onboarding/OnboardIdentityView.js]: I agree, all of this has been rushed in and I don't think it's a good UX. I just think if we're saying Mainframe is "private by default", that's what our defaults should be, or at least we should make sure users are properly informed before publishing any data. But that's fine we don't have time anyways so let's just push what we have.

**Pull request merged**